blocksync/pool.go

```diff
@@ -155,6 +211,42 @@ func (pool *BlockPool) makeRequestersRoutine() {
 	}
 }
 
+// dropExcessRequesters removes requesters that exceed the allowed limit.
+// Drops requesters from highest height to lowest (furthest from completion).
+// CONTRACT: pool.mtx must be locked.
+func (pool *BlockPool) dropExcessRequesters(maxRequestersAllowed int, maxPending int) {
+	numRequesters := len(pool.requesters)
+	if numRequesters <= maxRequestersAllowed {
+		return
+	}
+
+	// Find the highest height requesters to drop (they're furthest from completion)
+	heights := make([]int64, 0, numRequesters)
+	for height := range pool.requesters {
+		heights = append(heights, height)
+	}
+	sort.Slice(heights, func(i, j int) bool {
+		return heights[i] > heights[j] // sort descending
+	})
+
+	// Drop requesters from highest to lowest until we're at the limit
+	numToDrop := numRequesters - maxRequestersAllowed
+
+	for i := 0; i < numToDrop && i < len(heights); i++ {
+		height := heights[i]
+		if requester, ok := pool.requesters[height]; ok {
+			// Always track dropped requesters to avoid banning peers for late-arriving blocks
+			pool.droppedRequesters[height] = struct{}{}
+			pool.Logger.Trace("Tracking dropped requester", "height", height)
```

Comment:
should this `tracking` be `deleting` or disconnecting or something? since I see that this loop is deleting the requests

Reply:
Changed the comment